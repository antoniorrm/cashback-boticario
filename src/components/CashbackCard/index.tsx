import React from 'react'
import { makeStyles } from '@material-ui/core/styles'
import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'
import Typography from '@material-ui/core/Typography'

const useStyles = makeStyles({
	root: {
		minWidth: 275,
	},
})

const CashbackCard = () => {
	const classes = useStyles()

	return (
		<Card className={classes.root}>
			<CardContent>
				<Typography variant='h5' component='h2'>
					Cashback acumulado
				</Typography>
				<Typography variant='body2' component='p'>
					R$ 100,00
				</Typography>
			</CardContent>
		</Card>
	)
}

export default CashbackCard
