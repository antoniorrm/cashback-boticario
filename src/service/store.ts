/* eslint-disable import/prefer-default-export */
import { store } from '../store'

export function getAccessToken() {
	return store.getState().auth.token
}