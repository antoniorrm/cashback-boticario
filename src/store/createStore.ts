import { createStore, applyMiddleware, Middleware, Reducer } from 'redux'
import { PersistState } from 'redux-persist'
import { AuthAction, AuthState } from './modules/auth/types'

export interface StoreState {
	_persist: PersistState
	auth: AuthState
}

export type StoreAction = AuthAction

export default (
	reducers: Reducer<StoreState, StoreAction>,
	middlewares: Middleware[]
) => {
	const enhancer = applyMiddleware(...middlewares)

	return createStore(reducers, enhancer)
}
