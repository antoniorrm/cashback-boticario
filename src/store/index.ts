import createSagaMiddleware from 'redux-saga'
import { AnyAction, Middleware, Store } from 'redux'
import { persistStore } from 'redux-persist'

import createStore from './createStore'
import rootReducer from './modules/rootReducer'
import rootSaga from './modules/rootSaga'
import persistReducers from './persistedReducers'

const sagaMiddleware = createSagaMiddleware()

const middlewares: Middleware[] = [sagaMiddleware]

const store = createStore(persistReducers(rootReducer), middlewares)
const persistor = persistStore(store as unknown as Store<any, AnyAction>)

sagaMiddleware.run(rootSaga)

export { store, persistor }
