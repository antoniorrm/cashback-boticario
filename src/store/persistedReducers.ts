import storage from 'redux-persist/lib/storage'
import { persistReducer } from 'redux-persist'
import { Reducer } from 'redux'
import { StoreAction, StoreState } from './createStore'

export default (reducers: Reducer<StoreState, StoreAction>) => {
	const persistedReducers = persistReducer(
		{
			key: 'cashback',
			storage,
			whitelist: ['auth'],
		},
		reducers
	)

	return persistedReducers
}
