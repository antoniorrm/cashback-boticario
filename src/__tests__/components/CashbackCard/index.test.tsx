/* eslint-disable no-undef */
import React from 'react'
import { render, screen } from '@testing-library/react'
import { SnackbarProvider } from 'notistack'
import { Provider } from 'react-redux'
import CashbackCard from '../../../components/CashbackCard'
import { store } from '../../../store'

describe('CashbackCard render sucess', () => {
	it('renders component correct', () => {
		const { container } = render(
			<Provider store={store}>
				<SnackbarProvider maxSnack={1}>
					<CashbackCard />
				</SnackbarProvider>
			</Provider>
		)
		const testComponent = screen.getByText(/Cashback acumulado/i)
		expect(container).toMatchSnapshot()
		expect(testComponent).toBeInTheDocument()
	})
})
