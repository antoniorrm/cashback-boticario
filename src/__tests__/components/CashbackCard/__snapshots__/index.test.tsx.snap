// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CashbackCard render sucess renders component correct 1`] = `
<div>
  <div
    class="MuiPaper-root MuiCard-root makeStyles-root-1 MuiPaper-elevation1 MuiPaper-rounded"
  >
    <div
      class="MuiCardContent-root"
    >
      <h2
        class="MuiTypography-root MuiTypography-h5"
      >
        Cashback acumulado
      </h2>
      <p
        class="MuiTypography-root MuiTypography-body2"
      >
        R$Â 0,00
      </p>
    </div>
  </div>
</div>
`;
